### DISCUSSION TITLE
state feedback controller design

### DISCUSSION THREAD

**Discussion by Leona-kk** [CURENT/andes#465] [Modeling]
Hi Dr. Cui and other developers,

I try to add a state feedback controller block (LQR controller). 

Based on the Block - Gain and closed loop pi controller design, I try to develop a multi-input Gain and select rotor angle/ speed/ eq'/ efd as the control input and vi as a control output. (similar to what people usually do in Matlab Simulink). But it always has initialisation issues.  Would you mind giving me any suggestions?
<img width="556" alt="Screen Shot 2023-06-30 at 4 01 48 pm" src="https://github.com/CURENT/andes/assets/129546504/3eb8d150-11e6-4c1b-8880-c708fe871210">

Cheers,
Yichao

**Comment by cuihantao** [comment]
Yichao,

I can help you check your code. Feel free to send it to me at cuihantao@gmail.com.
